File: Server/src/main/content/global/skill/combat/summoning/familiar/npc/SpiritGraahkNPC.java
package content.global.skill.combat.summoning.familiar.npc;

import content.global.skill.combat.summoning.familiar.Familiar;
import content.global.skill.combat.summoning.familiar.FamiliarSpecial;
import core.game.node.entity.combat.equipment.WeaponInterface;
import core.game.node.entity.player.Player;
import core.game.node.entity.skill.SkillBonus;
import core.game.node.entity.skill.Skills;
import core.plugin.Initializable;

@Initializable
public class SpiritGraahkNPC extends Familiar {

    public SpiritGraahkNPC() {
        this(null, 7363);
    }

    public SpiritGraahkNPC(Player owner, int id) {
        super(owner, id, 4900, 12810, 3, WeaponInterface.STYLE_AGGRESSIVE);
        boosts.add(new SkillBonus(Skills.HUNTER, 5));
    }

    @Override
    public Familiar construct(Player owner, int id) {
        return new SpiritGraahkNPC(owner, id);
    }

    @Override
    protected boolean specialMove(FamiliarSpecial special) {
        if (!super.isOwnerAttackable()) {
            return false;
        }
        call();
        return true;
    }

    @Override
    public int[] getIds() {
        return new int[]{7363, 7364};
    }

}

File: Server/src/main/content/global/skill/combat/summoning/familiar/npc/SpiritGraahkNPC.kt
package content.global.skill.combat.summoning.familiar.npc

import content.global.skill.combat.summoning.familiar.Familiar
import content.global.skill.combat.summoning.familiar.FamiliarSpecial
import core.game.node.entity.combat.equipment.WeaponInterface
import core.game.node.entity.player.Player
import core.game.node.entity.skill.SkillBonus
import core.game.node.entity.skill.Skills
import core.plugin.Initializable

@Initializable
class SpiritGraahkNPC @JvmOverloads constructor(owner: Player? = null, id: Int = 7363) :
    Familiar(owner, id, 4900, 12810, 3, WeaponInterface.STYLE_AGGRESSIVE) {
    init {
        boosts.add(SkillBonus(Skills.HUNTER, 5.0))
    }

    override fun construct(owner: Player, id: Int): Familiar {
        return SpiritGraahkNPC(owner, id)
    }

    override fun specialMove(special: FamiliarSpecial): Boolean {
        if (!super.isOwnerAttackable()) {
            return false
        }
        call()
        return true
    }

    override fun getIds(): IntArray {
        return intArrayOf(7363, 7364)
    }
}
